nmap_test.go

```diff
@@ -287,7 +287,7 @@ func TestRunWithProgress(t *testing.T) {
 
 			compareWholeRun:  true,
 			expectedResult:   r,
-			expectedProgress: []float32{56.66, 81.95, 87.84, 94.43, 97.76, 97.76},
+			expectedProgress: []float32{3.22, 56.66, 77.02, 81.95, 86.79, 87.84},
```

Comment:
Why did this change? 🤔

Reply:
This line changed by improving (speeding up) the [script](https://github.com/Ullaakut/nmap/blob/77f28efd3b74bbb89a9506ffcad732e40648ed0d/tests/scripts/fake_nmap_delay.sh) that is running in this test.

- The reason is that the function `RunWithProgress` is _polling_ the progress of a nmap-run.
- Therefore it is very difficult to write a test that captures all the xml progress tags and push them through the channel.